src/wallet/api.rs

```diff
@@ -1708,23 +1708,26 @@ impl Wallet {
         Ok(swept_txids)
     }
 
-    /// Waits for a sweep transaction to confirm and returns its block height.
-    pub fn wait_for_sweep_tx_confirmation(&self, txid: Txid) -> Result<u32, WalletError> {
+    /// Waits for a transaction to confirm and returns its block height.
+    pub(crate) fn wait_for_tx_confirmation(&self, txid: Txid) -> Result<u32, WalletError> {
         let sleep_increment = 10;
         let mut sleep_multiplier = 0;
 
         let ht = loop {
             sleep_multiplier += 1;
+
             let get_tx_result = self.rpc.get_transaction(&txid, None)?;
             if let Some(ht) = get_tx_result.info.blockheight {
+                log::info!("Transaction {txid} confirmed at blockheight: {ht}");
                 break ht;
             } else {
-                log::info!("Sweep Transaction {txid} seen in mempool, waiting for confirmation.");
+                log::info!("Transaction {txid} seen in mempool, waiting for confirmation.");
```

Comment:
We can turn this into "variable at end" pattern then we can await for the whole thing in the log.

`Transaction confirmed at height: {}, txid: {}` and `Transaction seen in mempool, waiting for confirmation, txid: {}`.